### PR TITLE
Prevent multiple kick/ban of 1 player

### PR DESCRIPTION
**Pull request by CrosRoad95**
This PR disallow kick/ban player if player was kicked/banned

Currently code below will display hunders of information `player was kicked bla bla bla`

Also i added check which return `false` when you try redirect player in `onPlayerQuit` event

script serverside
```lua
function quitPlayer ( quitType )
	iprint("kick 2", kickPlayer(source))
end
addEventHandler ( "onPlayerQuit", getRootElement(), quitPlayer )
iprint("ban 1", kickPlayer(getRandomPlayer()))

### CODE REVIEW

File: Server/mods/deathmatch/logic/CPlayer.h [hunk 1]
```diff
@@ -351,6 +352,7 @@ class CPlayer : public CPed, public CClient
     uint                                        m_uiD3d9Size;
     SString                                     m_strD3d9Md5;
     SString                                     m_strD3d9Sha256;
+    bool                                        bIsQuit;
```
Comment (Necktrox) [inline Server/mods/deathmatch/logic/CPlayer.h]: Member variables must be prefixed with **m_**. Furthermore, you might want to name this variable **m_bIsLeavingServer**, but this is up for debate.

File: Server/mods/deathmatch/logic/CPlayer.h [hunk 2]
```diff
@@ -276,6 +276,7 @@ class CPlayer : public CPed, public CClient
 
     CVehicle *                                  GetJackingVehicle           ( void )                        { return m_pJackingVehicle; }
     void                                        SetJackingVehicle           ( CVehicle * pVehicle );
+    bool                                        IsQuit                      ( void )                        { return bIsQuit; }
```
Comment (Necktrox) [inline Server/mods/deathmatch/logic/CPlayer.h]: You might want to name this method **IsLeavingServer**, but this is up for debate.

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 3]
```diff
@@ -11473,7 +11477,9 @@ CBan* CStaticFunctionDefinitions::BanPlayer ( CPlayer* pPlayer, bool bIP, bool b
 {
     // Make sure we have a player
     assert ( pPlayer );
-
+    if ( pPlayer->IsQuit() ) {
+        return NULL;
```
Comment (Necktrox) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Return **nullptr** instead.

**Comment by CrosRoad95** [comment]
```lua
addEventHandler ( "onPlayerQuit", getRootElement(), function()
  iprint("kick, redirect in onPlayerQuit event", kickPlayer(source), redirectPlayer(source,"",0)) -- return false, false
end )

plr = getRandomPlayer()
if(plr)then
  iprint("kick outside event", kickPlayer(getRandomPlayer()))
end

addEventHandler ("onPlayerJoin", root, function()
  iprint("kick in onPlayerJoint event", kickPlayer(source)) -- return true
end)
```
dj-k9uzuqj2luslajlubxq

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 4]
```diff
@@ -11141,8 +11145,11 @@ CBan* CStaticFunctionDefinitions::BanPlayer(CPlayer* pPlayer, bool bIP, bool bUs
                                             SString strReason, time_t tUnban)
 {
     // Make sure we have a player
-    assert(pPlayer);
 
+    assert (pPlayer);
+    if ( pPlayer->IsLeavingServer() ) {
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: The ban here should still be applied

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 5]
```diff
@@ -11095,8 +11097,10 @@ bool CStaticFunctionDefinitions::LogOut(CPlayer* pPlayer)
 bool CStaticFunctionDefinitions::KickPlayer(CPlayer* pPlayer, SString strResponsible, SString strReason)
 {
     // Make sure we have a player
-    assert(pPlayer);
-
+    assert (pPlayer);
+    if ( pPlayer->IsLeavingServer() ) {
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: please fix formatting (whitespace)

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 6]
```diff
@@ -3386,7 +3386,9 @@ bool CStaticFunctionDefinitions::RedirectPlayer(CElement* pElement, const char*
     if (IS_PLAYER(pElement))
     {
         CPlayer* pPlayer = static_cast<CPlayer*>(pElement);
-
+        if ( pPlayer->IsLeavingServer() ) {
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: please fix formatting (whitespace)

File: Server/mods/deathmatch/logic/CPlayer.h [hunk 7]
```diff
@@ -266,7 +266,7 @@ class CPlayer : public CPed, public CClient
 
     CVehicle* GetJackingVehicle(void) { return m_pJackingVehicle; }
     void      SetJackingVehicle(CVehicle* pVehicle);
-
+    bool      IsLeavingServer( void ){ return m_bIsLeavingServer; }
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CPlayer.h]: please fix formatting (whitespace)

File: Server/mods/deathmatch/logic/CGame.cpp [hunk 8]
```diff
@@ -1346,6 +1346,9 @@ void CGame::InitialDataStream(CPlayer& Player)
 
 void CGame::QuitPlayer(CPlayer& Player, CClient::eQuitReasons Reason, bool bSayInConsole, const char* szKickReason, const char* szResponsiblePlayer)
 {
+    if ( Player.m_bIsLeavingServer) {   // prevent multiple kick one player
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CGame.cpp]: please fix formatting (whitespace)

**Review by qaisjp — CHANGES_REQUESTED** [review]

**Review by qaisjp — CHANGES_REQUESTED** [review]
Thanks! The ban should still be applied on their way out, though.

**Pull request merged**